server/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -711,6 +711,13 @@ public boolean isES87TSDBCodecEnabled() {
         Property.Final
     );
 
+    public static final Setting<Boolean> LOGSDB_SORT_ON_HOST_NAME = Setting.boolSetting(
```

Comment:
Just to verify my understanding of why we need this setting. Is it for cases where the mapping does contain a `host.name` field but the user doesn't want it to be used for sorting purposes? I suppose that is to have parity with the previous behavior where you could disable sorting by `host.name` by manually specifying just `@timestamp` as a sort field. Did I get that right?

Nit: there's a discrepancy in the naming (`SORT_ON_HOST_NAME` vs `use_default_sort_config`). The latter seems a bit more generic in case we add more fields to the default sort config. But either way, I think it makes sense to keep the name of the config option and the constant aligned.